docs/smart_contracts.rst

```diff
@@ -0,0 +1,142 @@
+Raiden Network Smart Contracts
+==============================
+
+Smart Contract Architecture
+---------------------------
+
+TODO
+
+
+Deployment and Use of the Smart Contracts
+-----------------------------------------
+
+Deploying
+~~~~~~~~~
+
+.. highlight:: bash
+
+To deploy the smart contracts use the provided utility script in
+`tools/deploy.py`. It expects an unencrypted hex-encoded private key as it's
+only argument. The account of the provided private key needs to hold enough
+Ether to complete the deployment. At the time of writing a complete deployment
+requires approximately 5.7 million gas.
+Example::
+
+    python deploy.py --pretty 1111111111111111111111111111111111111111111111111111111111111111
+
+It will output a JSON formatted list of the deployed contracts / libraries
+addresses. You should keep this for the next steps.
+
+
+Using the Contracts
+~~~~~~~~~~~~~~~~~~~
+
+It's possible to have Raiden use other contracts than the default ones
+configured in `raiden/constants.py`. To do so use the following command line
+options:
+
+  - :code:`--discovery-contract-address` corresponds to `EndpointRegistry.sol`
+  - :code:`--registry-contract-address` corresponds to `Registry.sol`
+
+
+Verifying on Etherscan
+~~~~~~~~~~~~~~~~~~~~~~
+
+Due to the fact that the Raiden Network uses multiple contracts and libraries
+verifying them on Etherscan_ is a bit more involved.
+
+Generally it's important to use the `Contract Version 2.0 verifier`_. The
+previous version 1 will not work and produce confusing error messages.
+
+To verify the contract code on Etherscan_ follow these steps:
+
+#. Enter the :code:`Registry`'s contract address in the Etherscan search field
+#. Click "Contract Code"
+#. Modify the URL to :code:`verifyContract2` to use the new verification tool
+#. Enter the contract name (:code:`Registry`) and select the used solc version
+#. Paste the combined contract code into the provided text area
+   The combined contract code can be generated with the
+   :code:`tools/join-contracts.py` utility script::
+
+    cd raiden/smart_contracts
+    python ../../tools/join-contracts.py Registry.sol registry-joined.sol
+
+#. Enter :code:`ChannelManagerLibrary` and it's corresponding address into the
```

Comment:
Add a bullet point specifying `Optimized` or not and to say that by default we don't optimize at the moment